### ISSUE TITLE
Customizable URL for the info page

### ISSUE THREAD

**Issue by hho** [kubernetes-helm/chartmuseum#55]
I'm using chartmuseum in a Docker container behind a reverse proxy.

I've set the `--chart-url` option, so the generated URLs in the `index.yaml` are correct.

However, the info page (when accessing the domain with a browser) shows the client IP instead:

> To add this as a local chart repository, please run the following command:
> `helm repo add chartmuseum https://10.11.12.13`

It would be nice to have the URL in there also customizable.

Thanks!

**Comment by jdolitsky** [comment]
@hho We are using the [github.com/gin-contrib/location](https://github.com/gin-contrib/location) lib to determine this

Could you try setting the `X-Host` header on your requests passed to upstreams?

I'm seeing this in the code: https://github.com/gin-contrib/location/blob/911ef8299476b8a48e11ec0c26f2a0810345b690/config.go#L64

**Comment by hho** [comment]
That changed nothing. Again, the client's IP is shown.

My Go skills are nonexistent, but I think I figured out, what happens:

We're not even getting to [Line 64](https://github.com/gin-contrib/location/blob/911ef8299476b8a48e11ec0c26f2a0810345b690/config.go#L64), because the header is already set and it returns [a line earlier](https://github.com/gin-contrib/location/blob/911ef8299476b8a48e11ec0c26f2a0810345b690/config.go#L63). That field is set in [Line 32](https://github.com/gin-contrib/location/blob/911ef8299476b8a48e11ec0c26f2a0810345b690/config.go#L32), but it's using the `X-Forwarded-For` header (which is the client's IP!) and not the `X-Forwarded-Host` header, which would be correct.

So I think it's a bug in that library, and I wanted to open an issue – but it turns out, it has already been open for more than half a year: https://github.com/gin-contrib/location/issues/15 :disappointed: 

Closing, as this has to be fixed upstream.
